### PR TITLE
Use zipped files for `pubmed` source in `bbs_database parse` entrypoint

### LINKED ISSUE #583: Use zipped files for `Pubmed` articles in `bbs_database parse` entrypoint
After #570, we are assuming in `topic-extract` that the input is a `.xml.gz` file when the source is `PubMed`.
But in `parse` don't we assume we have the `xml` file (i.e. that the `.gz` was unzziped)?
https://github.com/BlueBrain/Search/blob/6e362760ef0d2bbb6cd583dfb363313d222737e2/src/bluesearch/entrypoint/database/parse.py#L130-L136

_Originally posted by @FrancescoCasalegno in https://github.com/BlueBrain/Search/pull/570#discussion_r802737633_

### PR DESCRIPTION
**Pull request by EmilieDel**
Fixes #583.

## Checklist

- [X] This PR refers to an issue from the [issue tracker](https://github.com/BlueBrain/Search/issues).
  (if it is not the case, please create an issue first).
- [X] Unit tests added.
  (if needed)
- [x] Documentation and `whatsnew.rst` updated.
  (if needed)
- [X] `setup.py` and `requirements.txt` updated with new dependencies.
  (if needed)
- [X] Type annotations added.
  (if a function is added or modified)
- [X] All CI tests pass.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -485,3 +485,14 @@ def mining_schema_df():
     schema_file = ROOT_PATH / "tests" / "data" / "mining" / "request" / "request.csv"
     df = pd.read_csv(schema_file)
     return df
+
+
+@pytest.fixture()
+def pubmed_articles_zipped_path(test_data_path, tmp_path):
```
Comment (Stannislav) [inline tests/conftest.py:491]: Up to you, but maybe one could use a shorter name? Seeing it causes line breaks down the line. Somethine like `pubmed_xml_gz` or similar?

File: tests/conftest.py [hunk 2]
```diff
@@ -485,3 +485,14 @@ def mining_schema_df():
...
+    pubmed_path = test_data_path / "pubmed_articles.xml"
+    zip_pubmed_path = tmp_path / "pubmed_articles.xml.gz"
+    with open(pubmed_path, "rb") as file_in, gzip.open(
+        zip_pubmed_path, "wb"
+    ) as gzip_out:
+        gzip_out.writelines(file_in)
```
[the remark this answers is not in the document]
Comment (EmilieDel) [reply tests/conftest.py:497]: I double-checked and it seems to work
```python
import gzip
from pathlib import Path

pubmed_path = Path("tests/data/pubmed_articles.xml")
zip_pubmed_path = Path("tests/data/pubmed_articles.xml.gz")

with open(pubmed_path, "rb") as file_in, gzip.open(zip_pubmed_path, "wb") as gzip_out:
    gzip_out.writelines(file_in)

with open(pubmed_path, "rb") as xml_stream:
    pubmed_article_xml = xml_stream.read()

with gzip.open(pubmed_path, "rb") as xml_stream:
    pubmed_article_zip = xml_stream.read()

print(pubmed_article_zip == pubmed_article_xml)
```

File: tests/conftest.py (hunk 1 above)
Comment (EmilieDel) [reply tests/conftest.py]: Done in [581588c](https://github.com/BlueBrain/Search/pull/582/commits/581588c91e530182c1d8f9c4a2e2dfac3ba5bd7b)

File: src/bluesearch/entrypoint/database/parse.py [hunk 3]
```diff
@@ -140,7 +141,9 @@ def iter_parsers(input_type: str, input_path: Path) -> Iterator[ArticleParser]:
         yield PubMedXMLParser(input_path)
 
     elif input_type == "pubmed-xml-set":
-        articles = ElementTree.parse(str(input_path))
+        with gzip.open(input_path) as mygzip:
+            articles_str = mygzip.read()
+        articles = ElementTree.fromstring(articles_str)
```
[the remark this answers is not in the document]
Comment (EmilieDel) [reply src/bluesearch/entrypoint/database/parse.py]: Done in [a19b0cf](https://github.com/BlueBrain/Search/pull/582/commits/a19b0cff145f06b6db67eeab3074f7755f0ee947)

**Review by jankrepl — APPROVED** [review]

**Pull request merged**